pages/api/users/index.ts

```diff
@@ -1,16 +1,44 @@
+import { PrismaClient, User } from '@prisma/client';
 import { NextApiRequest, NextApiResponse } from 'next';
-import { sampleUserData } from 'utils/sample-data';
+import sanitizeUser from 'utils/sanitizeUser';
+import UserSchema from 'interfaces/user';
+import CreateError, { MethodNotAllowed } from 'utils/error';
+import hash from 'utils/hashPassword';
 
-const handler = (_: NextApiRequest, res: NextApiResponse): void => {
-  try {
-    if (!Array.isArray(sampleUserData)) {
-      throw new Error('Cannot find user data');
-    }
+const prisma = new PrismaClient();
+
+export default async (
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<void> => {
+  // Index is only for POST request
+  if (req.method !== 'POST') {
+    return MethodNotAllowed(req.method, res);
+  }
+
+  // Construct payload
+  const registerInfo = {
+    firstName: req.body.firstName,
+    lastName: req.body.lastName,
+    email: req.body.email,
+    hashedPassword: hash(req.body.password), // I'm not sure why this has to be lowercase here
```

Comment:
make sure to handle where the fields don't exist in body

Reply:
actually, you should schema verify the `req.body` and then change password to hash